lib/s3.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const assert = require('assert');
+const awsSDK = require('aws-sdk');
+const logMeGently = require('@alpha-lambda/log-me-gently');
+const uuid = require('uuid');
+
+const AwsDriverError = require('./aws-driver-error');
+
+class S3Driver {
+  constructor({ AWS, client, conf = {}, level }) {
+    if (client) {
+      this._client = client;
+    } else {
+      const clientOptions = Object.assign({
+        apiVersion: '2006-03-01',
+        signatureVersion: 'v4'
+      }, conf.client);
+
+      const aws = AWS || awsSDK;
+      this._client = new aws.S3(clientOptions);
+    }
+
+    this._conf = conf;
+    this._logger = logMeGently({ level });
+  }
+
+  get client() {
```

Comment:
If you actually want a private varaible, you can use Object.defineProperties instead